Tokens management for workers

karitra opened the pull request:
subj.

TODO:
 - -[ ] ~~some test samples~~
 - [x] try to make service requests async

karitra committed:
feat(auth): tokens management for workers

3Hren commented on include/cocaine/framework/worker/options.hpp:
```diff
@@ -25,7 +25,19 @@ namespace cocaine {
+
 struct options_t {
+    // taken from environment variables
+    std::string token_type;
```
Unfortunately it breaks ABI. Use `other` member with public accessors instead.

karitra committed:
refactor(auth): move ticket options to options_t::other table

 - move ticket refresh corresponding fields from members to options_t::other mapping
 - try to use future.then(cb) on ticket service refresh request (still don't sure is it async?)

3Hren commented on include/cocaine/framework/worker/options.hpp:
```diff
@@ -25,6 +25,13 @@ namespace cocaine {
 namespace framework {
 
+namespace details {
+    constexpr auto TVM_TOKEN_TYPE = "TVM";
```
Move in `src`.

karitra committed:
fix(auth): prevent any_cast, at(name) exceptions

karitra commented:
Seems to work while using storage service as ticket provider mock, but don't yet have a clue how to make test cases more realistic, any advice? 

Also not sure that _async_ service call for `refresh_ticket` at ` tvm_service_impl_t::refresh_ticket_async` either correctly implemented or idiomatic. 

@3Hren, @antmat, PTAL.

3Hren commented on src/worker/options.cpp:
```diff
@@ -21,8 +21,22 @@
+    constexpr auto DEFAULT_TOKENS_SRV_NAME = "tvm";
+    constexpr auto DEFAULT_REFRESH_INTERVAL_SEC = 15u;
+
+    constexpr auto DEFAULT_TOKEN_BODY = "very_secret";
```
Leave it empty.

3Hren commented on include/cocaine/framework/worker/options.hpp:
```diff
@@ -41,6 +41,8 @@ struct options_t {
     protocol() const;
 
+    const boost::any&
+    at(const std::string& name) const;
```
Closer, but not what I wanted.

Look, we have predefined number of options passed from cocaine-runtime. Therefore we can describe them explicitly:

```c++
std::string token_type() const;
std::string token_body() const;
```

3Hren reviewed (changes requested):
LGTM except those minor issues I've mentioned.

karitra committed:
refactor(options): rearrange options_t

Rearrange options_t interface to fit its concept as predefined set of worker
parmeters. Methods added:
 - options_t::token_type
 - options_t::token_body

Remove/rearrange few redundant global constants.

3Hren commented:
LGTM, @antmat TAL.
